contracts/BIFI/strategies/QuickSwap/StrategyQuickswapDualRewardLP.sol

```diff
@@ -0,0 +1,305 @@
+        }
+
+        uint256 lp0Bal = IERC20(lpToken0).balanceOf(address(this));
+        uint256 lp1Bal = IERC20(lpToken1).balanceOf(address(this));
+        IUniswapRouterETH(unirouter).addLiquidity(lpToken0, lpToken1, lp0Bal, lp1Bal, 1, 1, address(this), now);
+    }
+
+    // calculate the total underlaying 'want' held by the strat.
+    function balanceOf() public view returns (uint256) {
+        return balanceOfWant().add(balanceOfPool());
+    }
+
+    // it calculates how much 'want' this contract holds.
+    function balanceOfWant() public view returns (uint256) {
+        return IERC20(want).balanceOf(address(this));
+    }
+
+    // it calculates how much 'want' the strategy has working in the farm.
+    function balanceOfPool() public view returns (uint256) {
+        return IRewardPool(rewardPool).balanceOf(address(this));
+    }
+
+    // returns rewards unharvested
+    function rewardsAvailable() public view returns (uint256) {
+        return IRewardPool(rewardPool).earned(address(this));
+    }
+
+    // returns native reward for calling harvest
+    function callReward() public view returns (uint256) {
+        uint256 outputBal = rewardsAvailable();
+        uint256 nativeOut;
+        if (outputBal > 0) {
+            try IUniswapRouterETH(unirouter).getAmountsOut(outputBal, outputToNativeRoute)
+            returns (uint256[] memory amountOut)
+            {
+                nativeOut = amountOut[amountOut.length - 1];
+            }
+            catch {}
+        }
+
+        return nativeOut.mul(45).div(1000).mul(callFee).div(MAX_FEE);
+    }
+
+    // called as part of strat migration. Sends all the available funds back to the vault.
+    function retireStrat() external {
+        require(msg.sender == vault, "!vault");
+
+        IRewardPool(rewardPool).withdraw(balanceOfPool());
+
+        uint256 wantBal = balanceOfWant();
+        IERC20(want).transfer(vault, wantBal);
+    }
+
+    function setHarvestOnDeposit(bool _harvestOnDeposit) external onlyManager {
+        harvestOnDeposit = _harvestOnDeposit;
+
+        if (harvestOnDeposit == true) {
+            super.setWithdrawalFee(0);
+        } else {
+            super.setWithdrawalFee(10);
+        }
+    }
+
+    // pauses deposits and withdraws all funds from third party systems.
+    function panic() public onlyManager {
+        pause();
+        IRewardPool(rewardPool).withdraw(balanceOfPool());
+    }
+
+    function pause() public onlyManager {
+        _pause();
+
+        _removeAllowances();
+    }
+
+    function unpause() external onlyManager {
+        _unpause();
+
+        _giveAllowances();
+
+        deposit();
+    }
+
+    function nativeToLp0() public view returns (address[] memory) {
+        return nativeToLp0Route;
+    }
+
+    function nativeToLp1() public view returns (address[] memory) {
+        return nativeToLp1Route;
+    }
+
+    function outputToNative() public view returns (address[] memory) {
```

Comment:
we need print `rewardToNative` as well.
i am more and more confident it will be easier to just remove 'reward' entity, but its ok we can adjust for it 😅